solana-programs/anchor/audius-data/tests/audius-data.ts

```diff
@@ -664,4 +664,155 @@ describe("audius-data", () => {
     ]);
     console.log(`Created 3 tracks in ${Date.now() - start}ms`);
   });
+
+  it("Save a track", async () => {
+    const { ethAccount, handleBytesArray, metadata } = initTestConstants();
+
+    const { baseAuthorityAccount, bumpSeed, derivedAddress: newUserAcctPDA } =
+      await findDerivedPair(
+        program.programId,
+        adminStgKeypair.publicKey,
+        Buffer.from(handleBytesArray)
+      );
+
+    // New sol key that will be used to permission user updates
+    const newUserKeypair = anchor.web3.Keypair.generate();
+
+    // Generate signed SECP instruction
+    // Message as the incoming public key
+    const message = newUserKeypair.publicKey.toString();
+
+    await testCreateUser({
+      provider,
+      program,
+      message,
+      baseAuthorityAccount,
+      ethAccount,
+      handleBytesArray,
+      bumpSeed,
+      metadata,
+      newUserKeypair,
+      userStgAccount: newUserAcctPDA,
+      adminStgPublicKey: adminStgKeypair.publicKey,
+    });
+
+    const newTrackKeypair = anchor.web3.Keypair.generate();
+    const trackMetadata = randomCID();
+    await testCreateTrack({
+      trackMetadata,
+      newTrackKeypair,
+      adminStgKeypair,
+      userAuthorityKeypair: newUserKeypair,
+      trackOwnerPDA: newUserAcctPDA,
+    })
+ 
+    const tx3 = await saveTrack({
+      program,
+      baseAuthorityAccount,
+      adminStgPublicKey: adminStgKeypair.publicKey,
+      userStgAccountPDA: newUserAcctPDA,
+      userAuthorityKeypair: newUserKeypair,
+      handleBytesArray,
+      bumpSeed,
+      trackAction: TrackActionEnumValues.save,
+      trackId: new anchor.BN("1"),
+    })
+
+    await confirmLogInTransaction(provider, tx3, 'success');
```

Comment:
can we parse out the argument here in the test and confirm whether it's a save or an unsave?

Reply:
the enum arg i mean